src/App.jsx

```diff
@@ -1,3 +1,53 @@
+import { useEffect, useState } from "react";
+import NewThoughtForm from "./NewThoughtForm";
+import HappyThought from "./HappyThought";
+
 export const App = () => {
-  return <div>Find me in src/app.jsx!</div>;
+  const [happyThoughts, setHappyThoughts] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetchThoughts();
+  }, []);
+
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch("https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts")
+      .then((response) => response.json())
+      .then((response) => setHappyThoughts(response))
+      .finally(() => setLoading(false));
+  };
+
+  const onNewThought = (newThought) => {
+    setHappyThoughts([newThought, ...happyThoughts]);
+  };
+
+  const onLike = (likedThought) => {
+    setHappyThoughts(
+      happyThoughts.map((happyThought) => {
+        if (happyThought._id === likedThought._id) {
```

Comment:
Great code here. Tried it like this in the beginning too but somehow broke my code. Feeling inspired now to try it next time again this way. 🌟